rskj-core/src/main/java/co/rsk/peg/P2shErpRedeemScriptBuilder.java

```diff
@@ -0,0 +1,82 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.Utils;
+import co.rsk.bitcoinj.core.VerificationException;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.bitcoinj.script.ScriptOpCodes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class P2shErpRedeemScriptBuilder implements ErpRedeemScriptBuilder{
+    private static final Logger logger = LoggerFactory.getLogger(P2shErpRedeemScriptBuilder.class);
+
+    public Script createRedeemScript(List<BtcECKey> defaultPublicKeys,
+                                     List<BtcECKey> emergencyPublicKeys,
+                                     long csvValue) {
+        Script defaultRedeemScript = ScriptBuilder.createRedeemScript(
+            defaultPublicKeys.size() / 2 + 1,
+            defaultPublicKeys);
+        Script emergencyRedeemScript = ScriptBuilder.createRedeemScript(
+            emergencyPublicKeys.size() / 2 + 1,
+            emergencyPublicKeys);
+
+        byte[] serializedCsvValue = Utils.signedLongToByteArrayLE(csvValue);
+        Script redeemScript = createRedeemScript(defaultRedeemScript, emergencyRedeemScript, serializedCsvValue);
+        validateRedeemScriptValues(defaultRedeemScript, emergencyRedeemScript, csvValue, redeemScript);
```

Comment:
```suggestion
   validateRedeemScriptValues(defaultRedeemScript, emergencyRedeemScript, csvValue);
        Script redeemScript = createRedeemScript(defaultRedeemScript, emergencyRedeemScript, serializedCsvValue);
        FederationUtils.validateScriptSize(redeemScript);
```
We should validate the values before creating the redeem script. It doesn't make sense for the method to receive all the separate values and also the redeem script itself. I think that's a different validation to be done after the redeem script is built.

Also, not sure about having that validation in a FederationUtils class. It's not really related to Federation but to scripts in general

Reply:
I agree, but the problem is that we dont know exactly the size of the redeem script before is created. We could do the math to estimate it (and probably would be pretty accurate if not exact), but this way we avoid mistakes.

According to the FederationUtils, you are right. I will change the file's name

Reply:
Good ones